core/src/plugins/filed/python/pyfiles/BareosFdWrapper.py

```diff
@@ -21,95 +21,157 @@
 #
 # Author: Maik Aussendorf
 #
-# The BareosFdWrapper module. Here is a global object bareos_fd_plugin_object
-# and wrapper functions, which are directly called out of the bareos-fd. They
-# are intended to pass the call to a method of an object of type
-# BareosFdPluginBaseclass (or derived)
 
-from bareosfd import bRC_OK, bRC_Error, JobMessage, M_FATAL
+"""The BareosFdWrapper module. Here is a global object bareos_fd_plugin_object
+and wrapper functions, which are directly called out of the bareos-fd. They
+are intended to pass the call to a method of an object of type
+BareosFdPluginBaseclass (or derived)"""
+
+# silence warning about module-name
+# pylint: disable=invalid-name
+# pylint: enable=invalid-name
+
+import traceback
+from functools import wraps
+from bareosfd import (  # pylint: disable=import-error
+    JobMessage,
+    DebugMessage,
+    bRC_OK,
+    bRC_Error,
+    M_FATAL,
+)
 
 # use this as global plugin object among your python-fd-plugin modules
-bareos_fd_plugin_object = None
-_plugin_class = None
+bareos_fd_plugin_object = None # pylint: disable=invalid-name
+PLUGIN_CLASS = None
 
+__all__ = ["bareos_fd_plugin_object"]
 
-def BareosPlugin(plugin_class):
-    global _plugin_class
-    _plugin_class = plugin_class
+
+def add_to_all(thing):
+    __all__.append(thing.__name__)
+    return thing
+
+
+@add_to_all
+def BareosPlugin(plugin_class): # pylint: disable=invalid-name
+    global PLUGIN_CLASS # pylint: disable=global-statement
+    PLUGIN_CLASS = plugin_class
     return plugin_class
 
 
+def bareos_catch_exceptions(wrapped_func):
+    @wraps(wrapped_func)
+    def wrap_func(*args, **kwargs):
+        try:
+            return wrapped_func(*args, **kwargs)
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            DebugMessage(20, f"{traceback.format_exc()}\n")
+            JobMessage(M_FATAL, f"{wrapped_func.__name__}() failed: {e}\n")
+            return bRC_Error
+
+    return wrap_func
```

Comment:
Do you have an example of a situation where an excepction is not caught by the `PyErrorHandler()` in the core ?

Reply:
I'm not sure the exception was not caught by `PyErrorHandler()`, but I remember having seen cases where there was no meaningful result.